netclient/ncutils/netclientutils.go

```diff
@@ -167,13 +168,15 @@ func GetPublicIP(api string) (string, error) {
 		client := &http.Client{
 			Timeout: time.Second * 10,
 		}
-		resp, err := client.Get(ipserver)
+		var resp *http.Response
+		resp, err = client.Get(ipserver)
 		if err != nil {
 			continue
 		}
 		defer resp.Body.Close()
 		if resp.StatusCode == http.StatusOK {
-			bodyBytes, err := io.ReadAll(resp.Body)
+			var bodyBytes []byte
+			bodyBytes, err = io.ReadAll(resp.Body)
```

Comment:
note that while this fixes the previous problem where one was handling an error outside of the for loop that wasn't in it's context - upon handling the error type outside of the for loop we're only going to know about the last error - not any of the others  
  
\#### tldr;   
it's better now but this for loop still has it's problems